### PR TITLE
Draft alignment spec for hit table

### PR DESCRIPTION
**Pull request by laserson**

### CODE REVIEW

**Review by javh — COMMENTED** [review]
Overall, looks good. I would probably simplify the names (eg, `sequence_id` to `id`, `segment_call` to `call`). But that's minor.

The only thing that might be a problem is how we want to handle different alignment scoring methods. We want to account for probabilistic aligners, but should we handle that with addition (optional) fields or by trying to generalize a few fields.  Ie, have `evalue` and `posterior` or just have `p` and leave the definition of `p` open to the aligner?

File: specs/alignments.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    type: float
+    doc: Alignment score
+    mandatory: True
+  -
+    name: segment_cigar
+    type: string
+    doc: Alignment CIGAR string
+    mandatory: True
+  -
+    name: segment_evalue
```
Comment (laserson) [inline specs/alignments.yaml]: Switch to `p` or probability.

Comment (laserson) [reply specs/alignments.yaml]: Actually, add "probability" and "log likelihood"

Comment (javh) [reply specs/alignments.yaml]: Also, maybe `rank`, at least as an optional column.

Comment (laserson) [reply specs/alignments.yaml]: @javh, we need all of `evalue`, `probability`, and `loglikelihood`?

Comment (laserson) [reply specs/alignments.yaml]: I'm only making `score` as "mandatory".  Also note that "mandatory" just means that the column must be reported in the TSV, but not that the values have to be filled

Comment (javh) [reply specs/alignments.yaml]: Probably? I know we'll need `evalue` for IgBLAST output and I suspect we'll need `probability` and `loglikelihood` for aligners like partis. @matsen and @psathyrella?

**Pull request merged**